ui/src/app/data/abstract-data.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from "@angular/core"
+import {HttpClient, HttpResponse} from "@angular/common/http"
+import { Router } from "@angular/router"
+import { Location } from "@angular/common"
+import { AuthService } from "../auth/auth-service"
+import {AbstractService, ApiGetParams} from "../abstract.service"
+import {Observable} from "rxjs"
+import {share} from "rxjs/operators"
+
+@Injectable({ providedIn: "root" })
+export abstract class AbstractDataService extends AbstractService {
+
+  protected constructor(httpClient: HttpClient, authService: AuthService, router: Router, location: Location) {
+    super(httpClient, authService, router, location)
+  }
+
+  /**
+   * Perform a delete request with a json response.  The resulting promise will return the whole
```

Comment:
Method descriptions should not include implementation details such as 'with a JSON response'. Return types should be self-evident by any @return types documentation or the method signature itself.